modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/unsafe.kt

```diff
@@ -0,0 +1,35 @@
+package arrow.effects
+
+import arrow.core.Continuation
+import java.util.concurrent.atomic.AtomicReference
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+import kotlin.coroutines.RestrictsSuspension
+import kotlin.coroutines.startCoroutine
+
+private class UnsafeContinuation<A>(
+  val result: AtomicReference<A> = AtomicReference()
```

Comment:
Use `Promise.uncancellable()` instead.

Reply:
Some other construct we have that's not AtomicReference then? I believe we had an alias somewhere.